superset-frontend/src/explore/controlUtils.js

```diff
@@ -121,19 +123,31 @@ function handleMissingChoice(control) {
   return control;
 }
 
-export function getControlStateFromControlConfig(controlConfig, state, value) {
+export function getControlStateFromControlConfig(
+  controlConfig,
+  controlPanelState,
+  value,
+) {
   // skip invalid config values
   if (!controlConfig) {
     return null;
   }
-  const controlState = applyMapStateToPropsToControl(
-    { ...controlConfig },
-    state,
-  );
+  const controlState = { ...controlConfig };
+  // only apply mapStateToProps when control states have been initialized
+  if (controlPanelState.controls || !controlPanelState.isInitializing) {
+    applyMapStateToPropsToControl(controlState, controlPanelState);
+  }
```

Comment:
you could make this something like:
```typescript
const controlStateWithControlPanelProps = controlPanelState.controls || !controlPanelState.isInitializing 
  ? applyMapStateToPropsToControl(controlState, controlPanelState)
  : controlState;
```

and then use it below. With that context, it should be fairly clear that we want to continue to use the latest created state variable. especially since the function isn't massive, I don't think people would mistakenly use the old state in this function

Reply:
I've changed it to use `let`. Not a big fan of having many variables. Always struggled with variable names and it quickly becomes confusing.